accepted/2020/workloads/workloads.md

```diff
@@ -0,0 +1,92 @@
+
+The target framework will be used as the primary currency in project files that establish a link to a workload. As part of a related effort, [target frameworks are being updated in .NET 5](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md) to include operating system information. This change will enable a direct link to OS-specific workloads. For example, a project that targets `net5-android` will need the .NET 5 Android workload (Xamarin.Android). A project that multi-targets `net5-android` and `net5-ios` will also need the .NET 5 iOS workload (Xamarin.iOS).
+
+The set of TFM to workload mappings will be finite, and defined by the workloads. The SDK will carry a copy of this mapping, but will defer to information carried as part of installed workloads when they present newer information.
+
+### Acquisition
+
+Workloads will be acquired in two primary ways: via a supported installation tool or the .NET CLI.
+
+Note: The term "supported installation tool" is intended to mean an installation system that exposes components and manages their dependencies. For example, Visual Studio Installer, Docker, and APT do that, while MSI and PKG don't.
+
+Install tools must follow these rules for workloads it installs:
+
+* The workload manifest is included as a required dependency for all variants of the workload (some packs might be optional).
+* All referenced packs are included, matching the version specified in the workload manifest.
+
+Users should be able to install workloads via an install tool without any sense that there is a compositional model at play.
+
+The .NET CLI will expose a more extensive set of commands for interacting with workloads. The CLI may allow installing more than just workloads, such that the noun it exposes is not `workload` but something more general (TBD; `bundle` is used as the stand-in for a better term in the following examples). In this parlance, a workload is one kind of bundle.
+
+* `dotnet bundle install [bundle]` -- Installs a bundle by name.
+* `dotnet bundle restore [project]` -- Installs one or more workloads in terms of the dependent workloads found in one or more referenced projects. One can think of restore installing the default bundle for a workload. The workload may expose other opt-in bundles.
```

Comment:
Could we merge `dotnet bundle restore` into `dotnet restore`?

Reply:
I agree. But we can also have a separate `dotnet sdk restore` to restore just the tools and sdks. useful for controlled env in scripting.

Reply:
No. There are a least two problems:

- A major goal is that we don't want workloads to participate in package graph calculation.
- We don't want restore to download gigabytes of content in any scenario. We think we need a new experience for large downloads.